testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -89,121 +209,267 @@ var beaconPathsAndObjects = map[string]metadata{
+				return errJsonCast
 			}
-			castedl, ok := lhouseResp.(*validator.GetAttesterDutiesResponse)
+			lResp, ok := l.(*apimiddleware.ProposerSlashingsPoolResponseJson)
 			if !ok {
-				return errors.New("failed to cast type")
+				return errJsonCast
 			}
-			if len(castedp.Data) == 0 ||
-				len(castedl.Data) == 0 ||
-				len(castedp.Data) != len(castedl.Data) {
-				return fmt.Errorf("attester data does not match, prysm: %d lighthouse: %d", len(castedp.Data), len(castedl.Data))
+			if pResp.Data == nil {
+				return errEmptyPrysmData
 			}
-			return compareJSONResponseObjects(prysmResp, castedl)
+			if lResp.Data == nil {
+				return errEmptyLighthouseData
+			}
+			return nil
 		},
 	},
-	"/beacon/headers/{param1}": {
-		basepath: v1MiddlewarePathTemplate,
-		params: func(_ string, e primitives.Epoch) []string {
-			slot := uint64(0)
-			if e > 0 {
-				slot = (uint64(e) * uint64(params.BeaconConfig().SlotsPerEpoch)) - 1
+	"/beacon/pool/voluntary_exits": {
+		basepath: v1PathTemplate,
+		params: func(_ string, _ primitives.Epoch) []string {
+			return []string{}
+		},
+		prysmResps: map[string]interface{}{
+			"json": &beacon.ListVoluntaryExitsResponse{},
+		},
+		lighthouseResps: map[string]interface{}{
+			"json": &beacon.ListVoluntaryExitsResponse{},
+		},
+		customEvaluation: func(p interface{}, l interface{}) error {
+			pResp, ok := p.(*beacon.ListVoluntaryExitsResponse)
+			if !ok {
+				return errJsonCast
 			}
-			return []string{fmt.Sprintf("%v", slot)}
+			lResp, ok := l.(*beacon.ListVoluntaryExitsResponse)
+			if !ok {
+				return errJsonCast
+			}
+			if pResp.Data == nil {
+				return errEmptyPrysmData
+			}
+			if lResp.Data == nil {
+				return errEmptyLighthouseData
+			}
+			return nil
+		},
+	},
+	"/beacon/pool/bls_to_execution_changes": {
+		basepath: v1PathTemplate,
+		params: func(_ string, _ primitives.Epoch) []string {
+			return []string{}
 		},
 		prysmResps: map[string]interface{}{
-			"json": &beacon.GetBlockHeaderResponse{},
+			"json": &beacon.BLSToExecutionChangesPoolResponse{},
 		},
 		lighthouseResps: map[string]interface{}{
-			"json": &beacon.GetBlockHeaderResponse{},
+			"json": &beacon.BLSToExecutionChangesPoolResponse{},
+		},
+		customEvaluation: func(p interface{}, l interface{}) error {
+			pResp, ok := p.(*beacon.BLSToExecutionChangesPoolResponse)
+			if !ok {
+				return errJsonCast
+			}
+			lResp, ok := l.(*beacon.BLSToExecutionChangesPoolResponse)
+			if !ok {
+				return errJsonCast
+			}
+			if pResp.Data == nil {
+				return errEmptyPrysmData
+			}
+			if lResp.Data == nil {
+				return errEmptyLighthouseData
+			}
+			return nil
 		},
 	},
-	// we want to test comma-separated query params
-	"/beacon/states/{param1}/validators?id=0,1": {
-		basepath: v1MiddlewarePathTemplate,
-		params: func(_ string, e primitives.Epoch) []string {
+	"/config/fork_schedule": {
+		basepath: v1PathTemplate,
+		params: func(_ string, _ primitives.Epoch) []string {
+			return []string{}
```

Comment:
do we need this params parameter at all? it's not used so maybe we just do a nil check and not fill it out.